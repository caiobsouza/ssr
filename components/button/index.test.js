/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index.js';

describe('<Button>', () => {
    it('should be same label as passed in props', () => {
      const app = shallow(<Button type="primary">label</Button>);
      expect(app.find('.btn').text()).toEqual('label');
    });

    it('should be same type as passed in props', () => {
      const app = shallow(<Button type="primary">label</Button>);
      expect(app.find('.btn').hasClass('primary'));
    });
  });

describe('With Snapshot Testing', () => {
    it('should be rendered', () => {
      const component = renderer.create(<Button type="primary">Label</Button>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });