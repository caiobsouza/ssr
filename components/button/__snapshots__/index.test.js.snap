// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing should be rendered 1`] = `
<a
  className="btn primary"
  role="button"
>
  Label
</a>
`;
